site/content/extend/plugins/best-practices.md

```diff
@@ -75,26 +75,45 @@ For examples of custom settings see: Demo Plugin [`CustomSetting`](https://githu
 
 ## How can I review the entire code base of a plugin?
 
-Carry out the following steps:
+Sometimes, you have been working on a personal repository for a new plugin, most probably based on the [mattermost-plugin-starter-template](https://github.com/mattermost/mattermost-plugin-starter-template/) repo. As it was a personal project, you may have pushed all of your commits directly commits to `master`. And now that it's functional, you need a reviewer to take a look at the whole thing.
 
-1. Take a backup of project directory
-2. Create a dummy-master branch with no code:
+For this, it is useful to create a PR with only the commits you added. Follow these steps to do so:
+
+0. Make sure that you are in the `master` branch, and that the remote is updated.
+
+   ```
+   git checkout master
+   git push origin master
+   ```
+   
+1. Create a new branch (it will be an exact copy of `master`) and push it to the remote:
+
+   ```
+   git checkout -b review
+   git push origin review
+   ```
+   
+2. Create another new branch and move to it. We will remove the new commits from this one so we can later create our PR:
+
+   ```
+   git checkout -b base
+   ```
+   
+2. Now, you need to obtain the identifier of the oldest commit that should be reviewed. You can do so with `git log --oneline`. From the output, copy the SHA of that specific commit. Let's say that the SHA is `8f6aef3`.
+3. Reset the new branch to the parent of the commit you identified before (the `~1` suffix will do the trick). Don't forget to change the SHA to the one you copied!
 
    ```
-   git checkout --orphan dummy-master
-   git rm -rf
-   git push origin dummy-master
+   git reset --hard 8f6aef3~1
```

Comment:
Instead of doing this with `HEAD`, I'd just propose a similar strategy to above, and write:
```
git branch base 8f6aef3~1
git push origin base
```

Reply:
Done in https://github.com/mattermost/mattermost-developer-documentation/pull/652/commits/c8ab06a7648c27f059aff8b68459af275a64d263.